drivers: pwm: Channel > 1 not working on rp2040 (Raspberry Pi Pico)

68ec020 opened the pull request:
pwm_set_chan_level uses slice channels A(=0) or B(=1) and not Zephyr channel (0..15). So PWM doesn't work for channels > 1. There is already a function (pwm_rpi_channel_to_pico_channel) which does the right thing, but it isn't used for pwm_set_chan_level.

yonsch reviewed (dismissed):
LGTM. Did you test this on a few PWM channels to make sure that it works?

68ec020 committed:
drivers: pwm: rpi_pico: Use “pico_channel” instead of “ch”

pwm_set_chan_level uses slice channels A(=0) or B(=1) and not Zephyr
channel (0..15). So PWM doesn't work for channels > 1. There is already
a function (pwm_rpi_channel_to_pico_channel) which does the right thing,
but it isn't used for pwm_set_chan_level.

Signed-off-by: Jan Hilsdorf <jan.hilsdorf@gmail.com>

68ec020 commented:
I just realised, I used my work email. And I didn't sign-off the commit. It is fixed (I hope) in the force-push.

68ec020 commented:
Sorry, another force push, to get the compliance checks fixed. I'm really sorry.

yonsch commented:
Don't worry, we all do that :)
Did you test it on a few channels to make sure it's working now?

68ec020 commented:
I tested it successfully on two pins (2 and 25). If you want, I can try it on all available pins (maybe except one pair of UART pins).

yonsch commented:
A few random pins should be fine
